rustc_llvm: remove stale references

tamird opened the pull request:
...that were removed in 77c3bfa7429abf87b76ba84108df018d9e9d90e2.

r? @alexcrichton

alexcrichton commented on src/librustc_llvm/ffi.rs:
```diff
@@ -505,17 +505,6 @@ pub mod debuginfo {
-// generates an llvmdeps.rs file next to this one which will be
-// automatically updated whenever LLVM is updated to include an up-to-date
-// set of the libraries we need to link to LLVM for.
-#[link(name = "rustllvm", kind = "static")] // not quite true but good enough
```
This line shouldn't be removed as I believe it's required for msvc

tamird replied:
Are you sure? cc::Build::compile prints to stdout a cargo directive that makes the link work, which I think is how this works. https://docs.rs/cc/1.0.3/src/cc/lib.rs.html#781

alexcrichton replied:
Yes, I am quite sure.

tamird replied:
Can you help me understand how hoedown, profiler-rt, binaryen_wrapper, and others work? AFAICT, none of them have `#[link(...)]` attributes.

alexcrichton replied:
Sure yes. This annotation is primarily needed for MSVC where attributes like dllimport/dllexport are applied and need to be correct for everything to link successfully. The `#[link]` annotation here says "these symbols are included statically" which means that they're all exported with `dllexport` and from the rustc_llvm dynamic library. Otherwise the rustc_trans dynamic library would not be able to access these symbols.

The other libraries we have just happen to not run into this situation where they cross dynamic library boundaries.

tamird replied:
Thanks! I've included your explanation in the code.

alexcrichton commented on src/librustc_llvm/ffi.rs:
```diff
@@ -505,17 +505,6 @@ pub mod debuginfo {
-// figure out the exact set of libraries. To do this, the build system
-// generates an llvmdeps.rs file next to this one which will be
-// automatically updated whenever LLVM is updated to include an up-to-date
-// set of the libraries we need to link to LLVM for.
```
Isn't this comment still largely applicable with some editing?

tamird replied:
Sort of, but there's no longer anything here to anchor it to, and the relevant code (in build.rs) is rather self explanatory.

alexcrichton replied:
Ok yeah on rereading let's leave out

alexcrichton commented:
@bors: r+

bors commented:
:pushpin: Commit 94d02b8 has been approved by `alexcrichton`

<!-- @bors r=alexcrichton 94d02b896c3feb5e997b95a660e850c7ad8cbe74 -->

bors commented:
:hourglass: Testing commit 94d02b896c3feb5e997b95a660e850c7ad8cbe74 with merge 24cbcaedb1f02525771187f5d944f23daa365e56...

bors commented:
:broken_heart: Test failed - [status-travis](https://travis-ci.org/rust-lang/rust/builds/308779646?utm_source=github_status&utm_medium=notification)

tamird commented:
Something to do with travis using an older xcode?
```
[00:13:17] In file included from /Users/travis/build/rust-lang/rust/src/llvm/include/llvm/Support/ThreadPool.h:30:
[00:13:17] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/future:1447:23: error: 'future_error' is unavailable: introduced in macOS 10.8
[00:13:17]                       future_error(make_error_code(future_errc::broken_promise))
[00:13:17]                       ^
[00:13:17] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/future:502:63: note: 'future_error' has been explicitly marked unavailable here
[00:13:17] class _LIBCPP_EXCEPTION_ABI _LIBCPP_AVAILABILITY_FUTURE_ERROR future_error
[00:13:17]                                                               ^
[00:13:17] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/future:1621:23: error: 'future_error' is unavailable: introduced in macOS 10.8
[00:13:17]                       future_error(make_error_code(future_errc::broken_promise))
[00:13:17]                       ^
[00:13:17] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/future:502:63: note: 'future_error' has been explicitly marked unavailable here
[00:13:17] class _LIBCPP_EXCEPTION_ABI _LIBCPP_AVAILABILITY_FUTURE_ERROR future_error
[00:13:17]                                                               ^
[00:13:17] 2 errors generated.
```
Doesn't seem related to this PR.

kennytm commented:
@bors retry — travis-ci/travis-ci#8821

shepmaster commented:
@bors retry
